### DISCUSSION TITLE
Custom Validation Rule conflicts with `RequiredRuleInferrer`

### DISCUSSION THREAD

**Discussion by juliomotol-hdmd** [spatie/laravel-data#53] [Q&A] [answered]
I have the following custom validation rule which uses the new `Enum` rule:

```php
use Attribute;
use Illuminate\Validation\Rules\Enum as EnumRule;
use Spatie\LaravelData\Attributes\Validation\ValidationAttribute;

#[Attribute(Attribute::TARGET_PROPERTY)]
class Enum extends ValidationAttribute
{
    public function __construct(
        private string $enumClass
    ) {
    }

    public function getRules(): array
    {
        return [new EnumRule($this->enumClass)];
    }
}
```

In the `getRules()` it returned and array of object.

When the `RequiredRuleInferrer` does its checks:

https://github.com/spatie/laravel-data/blob/eeabbc5a8401dddf579cb578ebe9a05da9305e05/src/RuleInferrers/RequiredRuleInferrer.php#L30-L32

It throws the ff:
> TypeError: str_starts_with(): Argument #1 ($haystack) must be of type string, Illuminate\Validation\Rules\Enum given in /home/vagrant/code/laravel/insurance-microservice/vendor/spatie/laravel-data/src/RuleInferrers/RequiredRuleInferrer.php:30

**Comment by rubenvanassche** [comment, marked as answer]
Hi @juliomotol-hdmd,

This issue should be fixed with v1.2.5, we've also added a Enum validation attribute by default!

**Reply by juliomotol-hdmd** [reply]
Great! It may be asking too much but could we also get an `EnumCast` as well? 🤣

**Reply by juliomotol** [reply]
I took the liberty of opening a PR for this, see https://github.com/spatie/laravel-data/pull/64.

**Reply by juliomotol** [reply]
@rubenvanassche this broke again in the 1.3.0 release. it now throws:

> Error: Object of class Illuminate\Validation\Rules\Enum could not be converted to string in /home/vagrant/code/laravel/insurance-microservice/vendor/spatie/laravel-data/src/Resolvers/DataPropertyValidationRulesResolver.php:59

See this [breaking diff](https://github.com/spatie/laravel-data/compare/1.2.5...1.3.0#diff-d0af6aff0184b05faacaab89f4d1e69a07d91e0ba5de1b24f4b4becfd1618bd7R54-R60)
